packages/components/copybutton/src/CopyButton.tsx

```diff
@@ -0,0 +1,109 @@
+import { cx } from 'emotion';
+import React, { useState, useCallback, useRef } from 'react';
+import CopyToClipboard from 'react-copy-to-clipboard';
+import { CommonProps } from '@contentful/f36-core';
+import { Copy } from '@contentful/f36-icons';
+import { Tooltip } from '@contentful/f36-tooltip';
+import { Box } from '@contentful/f36-core';
+import type { TooltipProps } from '@contentful/f36-tooltip';
+import { getStyles } from './CopyButton.styles';
+
+const DEFAULT_TAG = 'div';
+
+export interface CopyButtonProps extends CommonProps {
+  /**
+   * Function that gets called when the button is clicked
+   */
+  onCopy?: (string) => void;
+  /**
+   * Text to be shown when the button is clicked
+   * @default Copied!
+   */
+  tooltipCopiedText?: string;
+  /**
+   * Text to be shown when button is hovered or focused
+   * @default Copy to clipboard
+   */
+  tooltipText?: string;
+  /**
+   * Props that are passed to the tooltip component
+   */
+  tooltipProps?: Omit<TooltipProps, 'content' | 'children'>;
+  /**
+   * HTML element used to wrap the CopyButton
+   * @default div
+   */
+  as?: React.ElementType;
```

Comment:
Do we need `as` functionality here? I'd say no. As for me, it should always be a `button`